src/model/ScheduleDay.cpp

```diff
@@ -369,6 +447,7 @@ namespace model {
     void ScheduleDay_Impl::clearCachedVariables() {
       m_cachedTimes.reset();
       m_cachedValues.reset();
+      m_cachedTimeSeries.reset();
```

Comment:
@jmarrec We need to somehow call this reset whenever the `numberOfTimestepsPerHour` changes on the `Timestep` object.

Reply:
Easiest is probably just to do it inside Timestep::setNumberOfTimestepsPerHour and resetNumberOfTimestepsPerHour.

Make ScheduleDay_Impl friend the Timestep_Impl.

```
    bool Timestep_Impl::setNumberOfTimestepsPerHour(int numberOfTimestepsPerHour) {
       if oldValue != numberOfTimestepsPerHour:
           for (auto& sch_day: model.getConcreteModelObjects<ScheduleDay>()) { clearCachedVariables }
      setInt as usual
```

Reply:
MMMH, you may need to make ScheduleDay (public) have a protected clearCachedVariables that forward to _Impl.

But getImpl is public I **think**


           for (auto& sch_day: model.getConcreteModelObjects<ScheduleDay>()) { sch_day.getImpl<ScheduleDay_Impl>()->clearCachedVariables() }

Reply:
Yep https://github.com/NREL/OpenStudio/blob/ab2d186f5afe7e95da38a26f87b8cf6f495ff64a/src/utilities/idf/IdfObject.hpp#L361-L370

Reply:
you need to make      `void clearCachedVariables();`protected instead of private, and friend Timeseries_Impl